src/Service/ZGWToXxllncService.php

```diff
@@ -293,13 +385,71 @@ public function mapZGWToXxllnc(string $casetypeId, ObjectEntity $zaakTypeObject,
     }//end mapZGWToXxllnc()
 
 
+    /**
+     * Maps zgw zaaktype to xxllnc casetype.
+     *
+     * @param string|null  $casetypeId          The caseType id.
+     * @param ObjectEntity $zaakTypeObject      ZGW ZaakType object.
+     * @param array        $zaakTypeArrayObject ZGW ZaakType object.
+     *
+     * @return array caseType array.
+     */
+    public function mapZGWZaakTypeToXxllnc(?string $casetypeId = null, ObjectEntity $zaakTypeObject, array $zaakTypeArrayObject): array
+    {
+        $mapping = $this->resourceService->getMapping('https://development.zaaksysteem.nl/mapping/xxllnc.XxllncZaakTypeToCaseType.mapping.json', 'xxllnc-zgw-bundle');
+        if ($mapping instanceof Mapping === false) {
+            return [];
+        }
+
+        $zaakTypeArrayObject = array_merge($zaakTypeArrayObject, [
+            'casetypeUuid' => $casetypeId,
+            'casetypeVersionUuid' =>  Uuid::uuid4()->toString(),
+            'catalogFolderUuid' => $this->configuration['catalogFolderUuid']
+        ]);
+
+        if (isset($zaakTypeArrayObject['roltypen']) === true) {
+            foreach ($zaakTypeArrayObject['roltypen'] as $key => $roltype) {
+                if (is_array($roltype) === true) {
+                    // Backup uuid is for when creating a rol in zaaksysteem (u have to give uuid instead they generate it).
+                    $zaakTypeArrayObject['roltypen'][$key]['backupUuid'] = Uuid::uuid4()->toString();
+                }
+            }
+        }
```

Comment:
Is er een casus dat roltypen niet bestaat? Normaal zou ik verwachten dat het een empty array is, en dan skip je hem al met je foreach (foreach ([] as $key => $value) {} komt neer op 'skip deze code').